### ISSUE TITLE
Future of this project? 

### ISSUE THREAD

**Issue by themauveavenger** [SheetJS/js-xlsx#370]
There haven't been any commits since April of last year, ton of open issues. Was wondering if this project is abandoned.

If so, can someone point me at a better alternative?

**Comment by SheetJSDev** [comment]
@themauveavenger we're still alive :) It's been a very long time since we've pushed a commit, but we've been looking at each of the issues and trying to find a solution that works across supported file formats and relevant versions of Excel.  Microsoft provides some documentation but oftentimes even Excel fails to generate files conforming to the docs.  But that's not what you or I or what anyone else sees.  We just see a file that works in Excel but fails with the command line tool or a web tool.

As an example of undocumented behavior, in Google Sheets try setting cell A1 to the number 1.3 and changing the format to `# ?/?` (fraction, up to 1 digit).  Do the same thing in Excel.  Google gives you "1 2/7" but Excel gives you "1 1/3".  After lots of digging and experimentation, it appears that Google is using the Mediant Method but Excel is using a completely different technique (the implementation of both methods lives in its own repo https://github.com/SheetJS/frac).  This isn't actually documented anywhere, and the fact that other applications don't implement this correctly means it's not immediately obvious.  In fact, the only reason I stumbled across this was in trying to design a test file for the fraction implementation.

It gets worse when you deal with third party apps that people trust.  In the process of designing the aforementioned test file, we discovered [bugs in LibreOffice](https://bugs.documentfoundation.org/show_bug.cgi?id=835101).  We were disappointed to find out that numerical accuracy wasn't a particularly high priority for them.  We've tried reporting bugs to relevant third party apps, but we get similar responses.  But as a user, from the outside you don't see that.  You just see that SheetJS is giving the "wrong" answer (and IMHO that's the right perspective).  It makes our task much more difficult because we have to accommodate for third parties that may not be doing the right thing.

I understand your frustration overall because none of this is obvious.  I didn't know or appreciate the spreadsheet data serialization mess at the beginning.  Excel is incredibly robust and does an excellent job of masking some issues in data files.  And sometimes you run into problems like https://github.com/PHPOffice/PHPExcel/issues/260#issuecomment-61432748, where it turns out that a tool that writes files is generating something that is valid according to Excel but not according to the official specification.  So before making a change, we should be mindful that data files come from a myriad of sources (Excel or a third party generator) and generated files will be used in various tools (Excel or a third party parser).  We should strive not to add to the problems that exist in the ecosystem.

There are certainly "better" alternatives (smaller source size, faster for just extracting text, more idiomatic to the programming language) out there in JS and in other languages.  In a day you could write something that reads and writes basic XLSX files, and you'd be able to tweak it to adjust for the files that you specifically care about (check out the References section in the README if this intrigues you).  It gets a lot more involved beyond that, and more often than not the implicit assumptions you make based on your [test corpus](https://github.com/SheetJS/test_files) turn out to be incorrect.

tl;dr: still alive, still working.

**Comment by hstonge** [comment]
Thanks for the update !

**Comment by pietersv** [comment]
One can only imagine the infinite number edge cases that led to writing 30,000+ unit test on an open source project.

**Comment by pietersv** [comment]
Is support for Node 4 / 5 in the works?

**Comment by SheetJSDev** [comment]
@pietersv that was pushed in 0.8.1 (node 0.12 end of maintenance was 12/31/2016 so we internally moved up to 6 and updated the module).  The 0.11 tests fail on travis and the 0.9 tests sometimes need to be restarted.  Currently [Travis](https://travis-ci.org/SheetJS/js-xlsx) is wired to test against node versions `0.8, 0.9, 0.10, 0.12, 4, 5, 6, 7`

**Comment by lgodard** [comment]
@SheetJSDev any update about  'bugs in LibreOffice' you pointed ?
the given URL points to nowhere

btw, keep up that good work :)

**Comment by SheetJSDev** [comment]
@lgodard here we go: https://bugs.documentfoundation.org/show_bug.cgi?id=83511

The original issue was reported in https://github.com/SheetJS/js-xls/issues/49#issuecomment-54542526

The LO developer response is truly horrifying:

> That is intentional. We ignore the last two bits for many stuff to improve the user experience.

IMHO fudging numbers is unacceptable for a program that people use for financials and other sensitive calculations.  After seeing this issue in particular, we started to advise people to move away from LibreOffice.
